### PR TITLE
add docs for workers

### LINKED ISSUE #1184: Question: Markdown, storage and security
Hi guys!

I realise this is a bug reporting section so I hope it's OK to post this, but I was hoping for some expert advice regarding Markdown and security. 

Firstly, my situation is that I want to allow members on our website to have a section of their user profile that they can edit using the markdown format, parsed by Marked.

Importantly, the data cannot be trusted - anyone can register for an account so I need to be careful with what can/can't be uploaded and displayed.

Here's how things currently stand on my development site:

1. I edit the raw parsedown code in a textarea and click submit.
2. The following is sent by POST to the server:

- DataA: marked([raw-unsanitized-data], { sanitize: true })
- DataB: The original raw data

3. DataA is put through [HTMLPurifier](https://github.com/ezyang/htmlpurifier) and then stored in a .htm file which is retrieved by http GET upon user profile load.
4. DataB is stored in the MySQL database as-is, as original data obviously required for when re-editing.
5. DataA (hopefully now sanitized) is returned and displayed on the user profile.

Hopefully that makes sense. Now, my question(s)..

Is that overkill? To the best of your knowledge is that safe enough?

In an idea world I'd like to cut out having to store the data twice and just work with the raw data *client-side* as the rendering seems to be blazingly fast - but from a security standpoint I don't know whether this is doable.

Main question: Would it be horribly irresponsible to just work with the raw data and just output:

`marked([raw-unsanitized-data], { sanitize: true })`

Possibly as an additional safeguard use [DOMPurify](https://github.com/cure53/DOMPurify):

`DOMPurify.sanitize(marked(<raw-unsanitized-data>, { sanitize: true }))`

? I can't help but feel like it probably would be, but I'd like to hear it from a real expert.

PS Further info, all site data is SSL and all cookies are secure/HttpOnly etc.

OK, if you're still here - thanks for staying awake!

But seriously, I'm no expert on this by any means so I really appreciate any insight/advice you can give.

Best Regards,

Rob

### PR DESCRIPTION
**Pull request by UziTech**
## Description

Add examples of how to run marked in a worker on node and in a browser.
https://docs-4nj255ohp.now.sh/#/USING_ADVANCED.md#workers

fixes #1184 

## Contributor

- [x] no tests required for this PR.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

File: docs/USING_ADVANCED.md [hunk 1]
```diff
@@ -64,15 +64,83 @@ console.log(myMarked('I am using __markdown__.'));
...
+```html
+<script>
+var markedWorker = new Worker('markedWorker.js');
+
+var markedTimeout = setTimeout(() => {
+  markedWorker.terminate();
+  throw new Error('Marked took too long!');
+}, timeoutLimit);
+
+markedWorker.onmessage = function (html) => {
```
Comment (styfle) [inline docs/USING_ADVANCED.md]:
```suggestion
markedWorker.onmessage = (html) => {
```

File: docs/USING_ADVANCED.md [hunk 2]
```diff
@@ -64,15 +64,83 @@ console.log(myMarked('I am using __markdown__.'));
...
+  clearTimeout(markedTimeout);
+  console.log(html);
+});
```
Comment (styfle) [inline docs/USING_ADVANCED.md]:
```suggestion
};
```

File: docs/USING_ADVANCED.md [hunk 3]
```diff
@@ -64,15 +64,83 @@ console.log(myMarked('I am using __markdown__.'));
...
+```
+
+### Web Worker
+
+```js
+// markedWorker.js
+
+importScripts('path/to/marked.min.js');
+
+onmessage = function (markdownString) {
```
Comment (styfle) [inline docs/USING_ADVANCED.md]:
```suggestion
onmessage = (markdownString) => {
```

File: docs/USING_ADVANCED.md [hunk 4]
```diff
@@ -14,12 +14,12 @@ marked(markdownString [,options] [,callback])
 
 ```js
 // Create reference instance
-var myMarked = require('marked');
+var marked = require('marked');
```
Comment (styfle) [inline docs/USING_ADVANCED.md]: Thank you! The `myMarked` variable has been annoying me for some time now 😉

**Comment by styfle** [comment]
There was one bug in the example that was mixing function keyword and arrow functions.

I think we should just use arrow functions for workers because its an advanced feature and so we can assume the reader understands ES6.

**Comment by UziTech** [comment]
Nice catch

**Comment by UziTech** [comment]
wait to merge this until after worker_threads in node are stable

**Comment by UziTech** [comment]
I added notes about Worker Threads being experimental and Web Workers sending objects on `postMessage` instead of just the string.

https://docs-4nj255ohp.now.sh/#/USING_ADVANCED.md#workers

I think this should be good to merge.

File: docs/USING_ADVANCED.md [hunk 5]
```diff
@@ -64,15 +64,93 @@ console.log(myMarked('I am using __markdown__.'));
...
+onmessage = (e) => {
+  const markdownString = e.data
+  postMessage(marked(markdownString));
+};
+```
+
+```js
+// script.js
+
+const markedWorker = new Worker('markedWorker.js');
```
Comment (styfle) [inline docs/USING_ADVANCED.md]:
```suggestion
const markedWorker = new Worker('./markedWorker.js');
```

**Pull request merged**